### ISSUE TITLE
Adding inline schema validation errors for text mode

### ISSUE THREAD

**Issue by meirotstein** [josdejong/jsoneditor#556]
Hi @josdejong !

Currently Schema validation errors are displayed at the bottom of the editor when you are in text mode.

IMHO, there are two problem with this approach: 

1. It is hard to determine the exact location of the error in the json (specially with big jsons)
2. It shows only the 3 first errors, if there are more than that you'll get "x more errors" message.

I have thought to add these errors on the side of the relevant lines, such as json parse errors.

Is the any technical reason why it wasn't implemented that way in the first place? if so - it will be better to be able to see the entire errors list - perhaps in a popup so it will not cover the entire editor.

Thanks!

**Comment by josdejong** [comment]
I would indeed be great if schema validation errors would be displayed inline in the code mode and text mode. 

This hasn't been done because it's simply quite some work to get that implemented, it's not trivial. The errors do not contain a line number, only the JSON path. We would have to translate that into the correct line number in the text editor. For the text editor we would have to implement a gutter and display the errors there. For the Ace editor we would have to create a new plugin or something like that to get the schema errors displayed in the gutter of Ace editor, not sure if that's possible and how it would work.

**Comment by meirotstein** [comment]
Got it - I will do some research.

I think that the latest work I have done with the cursor API might also be used here to determine the location of a json path in the text file.

As for the error display in the gutter:

For `code` mode we can use ACE Editor annotations, no need for new plugin, for example:
```js
editor.aceEditor.getSession().setAnnotations([{
  row: 1,
  column: 0,
  text: "Schema error",
  type: "warning"
}])
``` 
![image](https://user-images.githubusercontent.com/2209844/42870773-611cb950-8a81-11e8-9312-d2b847fd6d49.png)

For `text` mode I don't think we should aim for adding a gutter - here the current display is enough, we just have to provide a way to see the entire error list.

**Comment by josdejong** [comment]
wow that sounds like it is easy to integrate in Ace Editor as soon as we know on which lines to display the errors! 

We could of course also go for a partial solution: only implement such a solution for Ace mode and not for text mode, if implementation of a gutter is too much work for a first iteration.

**Comment by josdejong** [comment]
Fixed via #560, thanks a lot!
